### PR TITLE
Adds Defaulting Operator RFC

### PR DESCRIPTION
**Pull request by solomon-b**
RFC for Nullish operator.
[rendered](https://github.com/hasura/kriti-lang/blob/RFC/nullish-operator/RFCs/DefaultingOperator.md)

### CODE REVIEW

**Comment by sordina** [comment]
This makes a lot of sense and it's great that it's explicit rather than having a defaulting behaviour.

File: RFCs/NullishOperator.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+### Effects and Interactions
+
+When combined with the proposed Optional Chaining feature we will be
+able to do nice expressions like:
+
+```
+{{ $body?.bar.baz ?? $bar }}
+```
+
+Where we attempt to look up a nested value in `$body` and default to `$bar` if it is unbound or `null`.
```
Comment (sordina) [inline RFCs/NullishOperator.md:115]: Excellent!

**Review by sordina — APPROVED** [review]
I really don't see any drawbacks with this proposal. The only thing that might be worth examining is the precedence of the full set of operators to understand how they all compose if parens are omitted.

**Comment by jkachmar** [comment]
> The only thing that might be worth examining is the precedence of the full set of operators to understand how they all compose if parens are omitted.

+1 to this, i think this is probably a worthwhile thing to consider for whenever we add new operators with some fixity.

---

this seems relatively straightforward to me, and as with #29 we benefit from prior art within the JS ecosystem.

**Review by jkachmar — APPROVED** [review]

**Comment by solomon-b** [comment]
> I really don't see any drawbacks with this proposal. The only thing that might be worth examining is the precedence of the full set of operators to understand how they all compose if parens are omitted.

I agree we need to make precedence more explicit. I think we can do that in a separate PR, what do you think?

**Pull request merged**